From: Tetsu Yamamoto <tyamamot@redhat.com>
Date: Fri, 22 Aug 2008 11:19:25 -0400
Subject: [xen] ia64: fix ia64_leave_kernel
Message-id: 20080822151925.13356.94517.sendpatchset@pq0-1.lab.bos.redhat.com
O-Subject: [RHEL5.3 PATCH 2/3] xen-ia64: fix ia64_leave_kernel.
Bugzilla: 447453
RH-Acked-by: Bill Burns <bburns@redhat.com>
RH-Acked-by: Jarod Wilson <jwilson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

bz447453
# HG changeset patch
# User Isaku Yamahata <yamahata@valinux.co.jp>
# Date 1213077631 -32400
# Node ID b844f87db11debd27bf464bf6aff7cc42456d486
# Parent  8f2979e64af5ff3a333db678fa4fe4393b21273e
[IA64] fix ia64_leave_kernel.

This patch is for safe leaving hypervisor.
After calling do_softirq in ia64_leave_kernel, interrupts must be masked.

Signed-off-by: Akio Takebe <takebe_akio@jp.fujitsu.com>

diff --git a/arch/ia64/linux-xen/entry.S b/arch/ia64/linux-xen/entry.S
index 9f008c6..044d9b0 100644
--- a/arch/ia64/linux-xen/entry.S
+++ b/arch/ia64/linux-xen/entry.S
@@ -905,7 +905,7 @@ GLOBAL_ENTRY(ia64_leave_kernel)
 	;;
 (pUStk)	ssm psr.i
 (pUStk)	br.call.sptk.many b0=do_softirq
-(pUStk)	ssm psr.i
+(pUStk)	rsm psr.i
 	;;
 (pUStk)	br.call.sptk.many b0=reflect_event
 	;;
